### PR TITLE
update rbac v1beta1 to v1

### PR DESCRIPTION
**Pull request by kinarashah**
https://github.com/rancher/rancher/issues/35496

### CODE REVIEW

File: pkg/kontainer-engine/drivers/util/utils.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	v3 "github.com/rancher/rke/types"
 	"gopkg.in/yaml.v2"
 	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
```
Comment (prachidamle) [inline pkg/kontainer-engine/drivers/util/utils.go:12]: Do we need to keep both rbac/v1beta1 and rbac/v1 to support multiple k8s versions?

Comment (kinarashah) [reply pkg/kontainer-engine/drivers/util/utils.go:12]: I could be missing something, but I see both in 1.18 (the lowest version we need to support) - 

v1: https://github.com/kubernetes/kubernetes/blob/release-1.18/staging/src/k8s.io/api/rbac/v1/types.go 
v1beta1: https://github.com/kubernetes/kubernetes/blob/release-1.18/staging/src/k8s.io/api/rbac/v1beta1/types.go

**Review by ryansann — DISMISSED** [review]

File: pkg/kontainer-engine/drivers/util/utils.go (hunk 1 above)
Comment (snasovich) [reply pkg/kontainer-engine/drivers/util/utils.go:12]: JFYI here is a table I've compiled a while ago with old and replacement APIs and k8s versions they are available starting from ([internal SUSE link](https://confluence.suse.com/display/EN/Kubernetes+1.22+Checklist#Kubernetes1.22Checklist-k8s-1.22-removals)).

Comment (prachidamle) [reply pkg/kontainer-engine/drivers/util/utils.go:12]: Both is supported, but I wonder if we can just update everything to v1 if there is nothing tied to v1beta1 strictly? Or maybe I am missing something.

https://github.com/rancher/rancher/blob/f2b697bbc860213fc7720aa106fc6b0b6ecc33d2/pkg/kontainer-engine/drivers/util/utils.go#L129-L131
https://github.com/rancher/rancher/blob/f2b697bbc860213fc7720aa106fc6b0b6ecc33d2/pkg/kontainer-engine/drivers/util/utils.go#L90

Comment (snasovich) [reply pkg/kontainer-engine/drivers/util/utils.go:12]: @prachidamle , ah, misunderstood your original comment. Yes, I think we want to update those occurrences you found to v1 too and get rid of references to `k8s.io/api/rbac/v1beta1` altogether.

Comment (kinarashah) [reply pkg/kontainer-engine/drivers/util/utils.go:12]: Oh! yep I misunderstood too, I thought the question was if updating it to all to `v1` was safe. Will update everything to rbac v1

Comment (prachidamle) [reply pkg/kontainer-engine/drivers/util/utils.go:12]: cool, thanks

**Review by snasovich — APPROVED** [review]

**Review by prachidamle — APPROVED** [review]

**Comment by kinarashah** [comment]
Closing this because I pulled the commit in an existing PR and  and didn't want to wait on multiple drone builds to pass-https://github.com/rancher/rancher/pull/35357/commits/01053b646f5def7e7cadd11b7c5dd9e6d273227d. 

It's the exact change that was approved here, no net new changes.

**Pull request closed without merging**